registry/proxy/proxyauth.go

```diff
@@ -19,7 +19,7 @@ type credentials struct {
 }
 
 func (c credentials) Basic(u *url.URL) (string, string) {
-	up := c.creds[u.String()]
+	up := c.creds[tokenURL]
```

Comment:
See line 34 below. The map is being created with the value of the const `tokenURL`. That is likely a bug, considering that `remoteURL` is being passed it. Albeit, I am not entirely sure.

cc @dmcgowan